File: documentation/minimal.cpp
#include <mpi.h>
#include <asagi.h>
#include <iostream>

using namespace asagi;

int main(int argc, char** argv)
{
  MPI_Init(&argc, &argv);
  
  Grid* grid = Grid::create();
  
  if (grid->open("/path/to/netcdf/file.nc") != Grid::SUCCESS) {
    std::cout << "Could not load file" << std::endl;
    return 1;
  }
  
  std::cout << "Value at (0,0): " << grid->getFloat2D(0, 0) << std::endl;
  
  delete grid;
  
  MPI_Finalize();
  
  return 0;
}

File: documentation/minimal.f90
include 'asagi.inc'

program minimal
  use mpi
  use asagi
  implicit none
  
  integer :: grid_id
  integer :: error

  call MPI_Init( error )
  
  grid_id = grid_create( )

  if( grid_open( grid_id, "/path/to/netcdf/file.nc" ) /= GRID_SUCCESS ) then
    write (*,*) "Could not load file"
    call exit(1)
  end if

  write (*,*) "Value at (0,0):", grid_get_float( grid_id, 0.d+0, 0.d+0 )

  call grid_close( grid_id )

  call MPI_Finalize( error )
end program minimal